python/cli/convert/tum.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+"""
+Convert data from TUM "Euroc / DSO / ASL" benchmark format to Spectacular AI
+format. See <https://vision.in.tum.de/data/datasets/visual-inertial-dataset>
+for details about the data format.
+"""
+
+import argparse
+import csv
+import json
+import os
+from pathlib import Path
+import subprocess
+import yaml
+from contextlib import contextmanager
+import shutil
+import tempfile
+import tarfile
+import zipfile
+import numpy as np
+
+def define_args(parser):
+    parser.add_argument('input', help='Path to the input data in TUM format (.tar, .zip or directory)')
+    parser.add_argument('output', help='Path to the output directory')
+    parser.add_argument('--fps', type=int, default=20, help='Frames per second (metadata only)')
+    parser.add_argument('--crf', type=int, default=15, help='FFmpeg video compression quality (0=lossless)')
+
+def convertVideo(files, output, fps, crf):
+    # Use `-crf 0` for lossless compression.
+    subprocess.run(["ffmpeg",
+        "-y",
+        "-r", str(fps),
+        "-f", "image2",
+        "-pattern_type", "glob", "-i", files,
+        "-c:v", "libx264",
+        # "-preset", "ultrafast",
+        "-preset", "veryslow",
```

Comment:
Maybe `veryfast` could be used. I think it only/mainly affects the file size and not the quality, but the speed gain is huge.